lib/tesla.ex

```diff
@@ -546,4 +546,7 @@ defmodule Tesla do
 
   @doc false
   def encode_pair({key, value}), do: [{key, value}]
+
+  defp is_list_of_tuples?({k, _}) when is_binary(k), do: true
+  defp is_list_of_tuples?(_), do: false
 end
```

Comment:
Given how [`Keyword.keyword?`](https://github.com/elixir-lang/elixir/blob/v1.12.2/lib/elixir/lib/keyword.ex#L127) is implemented we could do the same:

```elixir
# on the top
if list_of_tuples?(value) do

# below

def list_of_tuples?([{k, _} | rest]) when is_atom(k) or is_binary(k), do: list_of_tuples?(rest)
def list_of_tuples?([]), do: true
def list_of_tuples?(_other), do: false
``

Reply:
Thanks, it looks much better.